tools/misbehaving-jmx-server/src/main/java/org/datadog/misbehavingjmxserver/App.java

```diff
@@ -82,28 +146,33 @@ public static void main( String[] args ) throws IOException, MalformedObjectName
             jCommander.usage();
             System.exit(1);
         }
+        //get bean count and properties from config file
+        config.initConfig();
 
         InterruptibleRMISocketFactory customRMISocketFactory = new InterruptibleRMISocketFactory();
         // I don't think this call is actually important for jmx, the below 'env' param to JMXConnectorServerFactory is the important one
         RMISocketFactory.setSocketFactory(customRMISocketFactory);
 
         // Explicitly set RMI hostname to specified argument value
-        System.setProperty("java.rmi.server.hostname", config.rmiHost);
-        
+        // This sometimes breaks containers it seems
+        if (config.rmiHost.equals("localhost")){
```

Comment:
lets revert this. If this is a fix that we want we should make it in a separate PR